python_modules/libraries/dagster-k8s/dagster_k8s/pipes.py

```diff
@@ -303,24 +305,54 @@ def build_pod_body(
             "A restart policy of Always is not allowed, computations are expected to complete."
         )
 
+    containers = spec["containers"]
+    init_containers = spec.get("init_containers") or []
+
     if "image" not in spec["containers"][0] and not image:
         raise DagsterInvariantViolationError(
             "Must specify image property or provide base_pod_spec with one set."
         )
 
-    if "name" not in spec["containers"][0]:
-        spec["containers"][0]["name"] = DEFAULT_CONTAINER_NAME
+    # We set the container name for the first container in the list if it is not set.
+    # There will be a validation error below for other containers.
+    if "name" not in containers[0]:
+        containers[0]["name"] = DEFAULT_CONTAINER_NAME
+
+    if not init_containers and len(containers) == 1:
+        if image:
+            containers[0]["image"] = image
+
+        if command:
+            containers[0]["command"] = command
+    else:
+        if image:
+            raise DagsterInvariantViolationError(
+                "Should specify 'image' via 'base_pod_spec' when specifying multiple containers"
+            )
+
+        if command:
+            raise DagsterInvariantViolationError(
+                "Should specify 'command' via 'base_pod_spec' when specifying multiple containers"
+            )
 
-    if image:
-        spec["containers"][0]["image"] = image
+        for container in (containers + init_containers):
+            for key in ["name", "image"]:
+                if key not in container:
+                    raise DagsterInvariantViolationError(f"Must provide base_pod_spec with {key} property set.")
```

Comment:
not critical: would be nice if the error message pointed you at which container / init_container had the problem, maybe just by index?